pepper-apis/src/main/java/org/broadinstitute/ddp/route/Auth0LogEventRoute.java

```diff
@@ -0,0 +1,139 @@
+ * Auth0 Custom Webhook settings (with prefix 'Bearer ').
+ */
+public class Auth0LogEventRoute implements Route {
+
+    private static final Logger LOG = getLogger(Auth0LogEventRoute.class);
+
+    /**
+     * It is possible to disable auth0 log event URL token check via system property
+     * (by default it is enabled and defined by config params).<br>
+     * If system property 'auth0LogEvent.tokenCheck.disable' = true then config params are ignored.
+     */
+    public static final String SYSTEM_PROPERTY_AUTH0_LOG_EVENT_TOKEN_CHECK_DISABLE = "auth0LogEvent.tokenCheck.disable";
+    /** By default token check is not disabled */
+    public static final boolean DEFAULT_AUTH0_LOG_EVENT_TOKEN_CHECK_DISABLE = false;
+
+    /** Mandatory parameter in the log event URL. Specifies name of auth0 tenant */
+    public static final String QUERY_PARAM_TENANT = "tenant";
+
+    private final Config config;
+
+    public Auth0LogEventRoute(final Config config) {
+        this.config = config;
+    }
+
+    @Override
+    public Object handle(Request request, Response response) throws Exception {
+        String tenant = readTenant(request);
+        checkAuthorizationToken(request);
+        checkBody(request);
+
+        final var auth0LogEventService = new Auth0LogEventService();
+
+        final var logEvents = auth0LogEventService.parseAuth0LogEvents(request.body());
+        for (final var logEvent : logEvents) {
+            final var logEventObject = Auth0LogEvent.createInstance(logEvent, tenant);
+            persistLogEvent(auth0LogEventService, logEventObject);
+            auth0LogEventService.logAuth0LogEvent(logEventObject);
+        }
+
+        response.status(SC_OK);
+        return "";
+    }
+
+    private String readTenant(final Request request) {
+        String tenant = request.queryParams(QUERY_PARAM_TENANT);
+        if (tenant == null) {
+            haltError(SC_BAD_REQUEST, REQUIRED_PARAMETER_MISSING, "Parameter not specified: " + QUERY_PARAM_TENANT);
+        }
+        return tenant;
+    }
+
+    private void checkAuthorizationToken(final Request request) {
+        if (isCheckToken()) {
+            String auth0LogEventApiBearerToken = config.getString(AUTH0_LOG_EVENT_BEARER_TOKEN);
+            if (isNotBlank(auth0LogEventApiBearerToken)) {
+                String authorizationToken = request.headers(AUTHORIZATION);
+                if (authorizationToken == null) {
+                    haltError(SC_BAD_REQUEST, REQUIRED_HEADER_MISSING, "Header not specified: " + AUTHORIZATION);
+                }
+                if (!addBearerPrefixToToken(auth0LogEventApiBearerToken).equals(authorizationToken)) {
+                    haltError(SC_UNAUTHORIZED, INVALID_TOKEN, "Invalid authorization token");
+                }
+            }
+        }
+    }
+
+    private boolean isCheckToken() {
+        return !SystemPropertyUtil.readProperty(SYSTEM_PROPERTY_AUTH0_LOG_EVENT_TOKEN_CHECK_DISABLE,
```

Comment:
Usually our system properties are read 100% by typesafe config, using `.withFallback()` as needed for defaults.  Can we remove `SystemPropertyUtil`, and pass in these values in the constructor?

Reply:
I thought may be to remove this system variable at all and in future define a paramter which could be set/unset via Google Cloud Runtime Configuration API ?

Reply:
Removed. The token check is disabled until the parameter not added to config file (now it is not there).